morphia/src/test/java/org/mongodb/morphia/TestQuery.java

```diff
@@ -369,37 +391,67 @@ public void testElemMatchQueryCanBeNegated() {
 
         Mapper mapper = getMorphia().getMapper();
 
-        List<Key<PhotoWithKeywords>> keys = getDs().find(PhotoWithKeywords.class)
+        validate(pwk1, pwk2, pwk3, pwk4, mapper, getDs().find(PhotoWithKeywords.class)
```

Comment:
Yes, I see that the validate methods are a refactoring.  But a bunch more assertions have been added to this test method and I'm suggesting that it would be better to restructure into multiple test methods whose named better describe what it is that is being tested.  And once you've done that, the test names you've chosen will suggest what the validate* methods should be renamed to to reflect their purpose.

Let me know if this doesn't make sense and we can talk about it more.